app/client/rest/tracking.ts

```diff
@@ -93,40 +99,118 @@ export default class ClientTracking {
         group.totalCompressedSize += metrics?.compressedSize ?? 0;
     }
 
-    incrementRequestCount(groupLabel: string) {
+    getAverageLatency(groupLabel: RequestGroupLabel): number {
+        const groupData = this.requestGroups.get(groupLabel);
+        if (!groupData) {
+            return 100;
+        }
+
+        const urlData = Object.entries(groupData.urls);
+        const sumLatency = urlData.reduce((result, url) => (result + (url[1].metrics?.latency ?? 0)), 0);
```

Comment:
Should we discard `0` latency from equation, ideally it should never happen though right?

Reply:
it should never happen really, the 0 here is forced to be type safe.